designs/Net5PlatformAndPlatformVersion.md

```diff
@@ -0,0 +1,177 @@
+# net5.0 TFM Platform and PlatformVersion implementation
+
+This document is meant to describe the various parts of the code that need to be updated in order to support platforms in net5.0 target framework identifiers. That is, the optional part after a `-` following `net5.0`. For example, `net5.0-ios14.1`. Note that this ONLY covers the platform part, because [initial support for `net5.0` has already been added](https://github.com/NuGet/NuGet.Client/pull/3265).
+
+For more details on the feature itself, please refer to [the full spec](https://github.com/terrajobst/designs/blob/master/accepted/2020/net5/net5.md).
+
+## class NuGetFramework
+
+Source: src/NuGet.Core/NuGet.Frameworks/NuGetFramework.cs
+
+### Fields
+
+Three additional fields will be added:
+
+```csharp
+        /// <summary>
+        /// Framework Platform (net5.0+)
+        /// </summary>
+        public string Platform
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// Framework Platform Version (net5.0+)
+        /// </summary>
+        public Version PlatformVersion
+        {
+            get;
+            private set;
+        }
+
+        /// <summary>
+        /// True if platform is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatform
+        {
+            get { return !string.IsNullOrEmpty(Platform); }
+        }
+
+        /// <summary>
+        /// True if platform version is non-empty (net5.0+)
+        /// </summary>
+        public bool HasPlatformVersion
+        {
+            get { return IsNet5Era && !(PlatformVersion.Major == 0 && PlatformVersion.Minor == 0); }

```

Comment:
I'm not sure the comment matches the logic, and there may be a better way of identifying "empty" versions. My understanding from the [Version Class](https://docs.microsoft.com/en-us/dotnet/api/system.version?view=netframework-4.8#remarks) is that -1 is the unassigned value, and 0.0 would be a valid version:

> The value of Version properties that have not been explicitly assigned a value is undefined (-1).

Reply:
Also, what's wrong will null? This is a valid use for null values. It seems wasteful to force the property to be not-null, and then invent our own convention for how to represent "not specified"

Reply:
The "no null" thing is something I pushed for. 

The reason is consistency. 
The platform version is expected to be declared in a similar way as the target framework version, see: https://nugettoolsdev.azurewebsites.net/5.5.0-preview.2.6382/parse-framework?framework=net. 

The serialization should be predictable too.
You'd get a v0.0 and that's a good enough experience.

If you have a TPI, you need a TPV. 

TPI needs a valid value to create a NuGetFramework, having a not declared TPV is weird. We need a default that makes more sense. 

0.0 is what TFV does, so we should do the same.

Reply:
yeah, this was a long conversation. If you still want to chat about it, we can jump on a call. Let me know, @zivkan and @donnie-msft!

Reply:
No need to chat, but maintain that the comment is inaccurate.